go/vt/vtgate/engine/set.go

```diff
@@ -371,84 +359,105 @@ func (svss *SysVarSetAware) MarshalJSON() ([]byte, error) {
 
 //Execute implements the SetOp interface method
 func (svss *SysVarSetAware) Execute(vcursor VCursor, env evalengine.ExpressionEnv) error {
+	var err error
 	switch svss.Name {
-	// These are all the boolean values we need to handle
-	case Autocommit, ClientFoundRows, SkipQueryPlanCache, TxReadOnly, TransactionReadOnly:
-		value, err := svss.Expr.Evaluate(env)
-		if err != nil {
-			return err
-		}
-		boolValue, err := value.ToBooleanStrict()
+	case sysvars.Autocommit.Name:
+		err = svss.setBoolSysVar(env, vcursor.Session().SetAutocommit)
+	case sysvars.ClientFoundRows.Name:
+		err = svss.setBoolSysVar(env, vcursor.Session().SetClientFoundRows)
+	case sysvars.SkipQueryPlanCache.Name:
+		err = svss.setBoolSysVar(env, vcursor.Session().SetSkipQueryPlanCache)
+	case sysvars.TxReadOnly.Name,
+		sysvars.TransactionReadOnly.Name:
+		// TODO (4127): This is a dangerous NOP.
+		noop := func(bool) error { return nil }
+		err = svss.setBoolSysVar(env, noop)
```

Comment:
if this required to perform svss.setBoolSysVar ?

Reply:
I call `setBoolSysVar` with a noop function just to check that the expression provided actually is a boolean. Otherwise, `SET @@transaction_read_only = 42` would not fail